Bump PHPStan analysis level

BelleNottelling opened the pull request:
This PR bumps our PHPStan config from level 1 to level 3.

This page describes what each level adds to the scanning:
https://phpstan.org/user-guide/rule-levels

For the most part, the changes I've made are to PHPDocs that were incorrect

Also, I did remove the logic that allowed a payment gateway to continue to function using the outdated `process` function as it added stupid complexity to the possible return value of a few of the internal functions.

The only one of ours that used that was the custom adapter and I've updated it to use the newer `processTransaction` function and anything else remaining likely has not been updated in at least a decade. So I think we are good to drop that code-path.

Anuril commented on src/library/FOSSBilling/Monolog.php:
```diff
@@ -51,7 +51,7 @@ public function __construct(protected ?\Pimple\Container $di)
 
             $formatter = new LineFormatter($this->outputFormat, $this->dateFormat, true, true, true);
-            $this->logger[$channel]->getHandlers()[0]->setFormatter($formatter);
+            $this->logger[$channel]->getHandlers()[0]->setFormatter($formatter); // @phpstan-ignore-line
```
// @phpstan-ignore-line Should be the absolute exeption, why is it used here? I think a comment explaining that would be good.

Anuril commented on src/library/Registrar/Adapter/Namecheap.php:
```diff
@@ -682,7 +682,7 @@ public function lock(Registrar_Domain $domain)
             throw new Registrar_Exception('Failed to :action: with the :type: registrar, check the error logs for further details', $placeholders);
         }
-        if ($result->CommandResponse->DomainGetRegistrarLockResult['IsSuccess'] == 'true') {
+        if ($result->CommandResponse->DomainGetRegistrarLockResult['IsSuccess'] == 'true') { // @phpstan-ignore-line
```
same here.

Anuril commented on src/library/Registrar/Adapter/Namecheap.php:
```diff
@@ -710,7 +710,7 @@ public function unlock(Registrar_Domain $domain)
             throw new Registrar_Exception('Failed to :action: with the :type: registrar, check the error logs for further details', $placeholders);
         }
-        if ($result->CommandResponse->DomainGetRegistrarLockResult['IsSuccess'] == 'true') {
+        if ($result->CommandResponse->DomainGetRegistrarLockResult['IsSuccess'] == 'true') { // @phpstan-ignore-line
```
and here.

Anuril commented on src/library/Server/Manager/Plesk.php:
```diff
@@ -133,6 +133,7 @@ public function deleteSubscription(Server_Account $a)
     public function suspendAccount(Server_Account $a, $suspend = true)
     {
     	if ($a->getReseller()) {
+            // @phpstan-ignore-next-line (This error is valid, however I don't have access to a plesk Instance to valudate if this request should just be made under `customer`)
```
valudate => Typo. Otherwise, this is what I'd think for the other exceptions

Anuril reviewed (commented):
Other than the lacking comments on the exception lines, LGTM

John-S4 commented:
Personally I could live with the lack of comments, but I do agree that maybe adding them will make things a bit clearer for anyone else looking at it.

BelleNottelling commented:
@John-S4 @Anuril 
Except for two cases where the code was correct as written, I've removed the ignores and added those errors to the baseline file.
PHPStan won't throw errors for them now, but having them in the baseline still means they are valid errors.

For the two pieces of code that are correct, I added a note to the ignore comment

John-S4 reviewed (approved):
Seems all good.

Anuril reviewed (approved):
LGTM
